controllers/backup.go

```diff
@@ -78,55 +82,333 @@ func (r *BackupReconciler) getActiveBackupName(backup *v1beta1.Backup, c client.
+	managedClusterList := clusterv1.ManagedClusterList{}
+	if err := c.List(ctx, &managedClusterList, &client.ListOptions{}); err != nil {
+		// if NotFound error
+		if !k8serr.IsNotFound(err) {
+			backupLogger.Info("managed clusters resources NOT FOUND")
+		} else {
+			backupLogger.Error(err, "failed to get clusterv1.ManagedClusterList")
+		}
+	} else {
+		for i := range managedClusterList.Items {
+			if managedClusterList.Items[i].Name == "local-cluster" {
+				continue
+			}
+			veleroBackup.Spec.IncludedNamespaces = appendUnique(veleroBackup.Spec.IncludedNamespaces, managedClusterList.Items[i].Name)
+		}
+	}
+}
+
+// update status for the last executed backup
+func updateLastBackupStatus(backup *v1beta1.Backup) {
+
+	if backup.Status.VeleroBackups != nil && len(backup.Status.VeleroBackups) > 0 {
+
+		backup.Status.Phase = v1beta1.StatusPhase(getBackupPhase(backup.Status.VeleroBackups))
+		if isBackupFinished(backup.Status.VeleroBackups) {
+			backup.Status.LastBackup = backup.Status.CurrentBackup
+			//backup.Status.CurrentBackup = ""
+			completedTime := getLastBackupCompletionTime(backup)
+			if completedTime != nil {
+				backup.Status.CompletionTimestamp = completedTime
+
+				startTime := backup.Status.VeleroBackups[0].Status.StartTimestamp
+				if startTime != nil {
+					duration := completedTime.Time.Sub(startTime.Time)
+					backup.Status.LastBackupDuration = getFormattedDuration(duration)
+
+				}
+			}
+		}
+	}
+}
+
+func isBackupFinished(backups []*veleroapi.Backup) bool {
+
+	if backups == nil || len(backups) <= 0 {
+		return false
+	}
+
+	// get all backups and check status for each
+	for i := 0; i < len(backups); i++ {
+		if backups[i].Status.Phase != "Completed" &&
+			backups[i].Status.Phase != "Failed" &&
+			backups[i].Status.Phase != "PartiallyFailed" {
+			return false // some backup is not ready
+		}
+	}
+
+	return true
+}
+
+// return cumulative status of backups
+func getBackupPhase(backups []*veleroapi.Backup) veleroapi.BackupPhase {
+
+	if backups == nil || len(backups) <= 0 {
+		return ""
+	}
+
+	// get all backups and check status for each
+	for i := 0; i < len(backups); i++ {
+
+		if backups[i].Status.Phase == "InProgress" {
+			return backups[i].Status.Phase // some backup is not ready, show that
+		}
+		if backups[i].Status.Phase == "Failed" {
+			return backups[i].Status.Phase // show failed first
+		}
+		if backups[i].Status.Phase == "PartiallyFailed" {
+			return backups[i].Status.Phase // show PartiallyFailed second
+		}
+	}
+
+	//if here, return the state of the first backup
+	return backups[0].Status.Phase
+}
+
+// return the execution time for the last backup in this list of backups ( creds, res, cluster)
+func getLastBackupCompletionTime(backup *v1beta1.Backup) *v1.Time {
+
+	if backup.Status.VeleroBackups == nil || len(backup.Status.VeleroBackups) <= 0 {
+		// no previous completed backup, can start one now
+		return nil
+	}
+
+	lastBackup := backup.Status.VeleroBackups[2]
```

Comment:
Can this always happen to have a backup at position [2]?

Reply:
changed with backup.Status.VeleroBackups[len(backup.Status.VeleroBackups)-1]